api/handlers.go

```diff
@@ -24,6 +24,16 @@ import (
 // APIVERSION is the API version supported by swarm manager
 const APIVERSION = "1.21"
 
+// A simple buffer reader to copy http.Request.Body stream
+type bufReader struct {
+	*bytes.Buffer
+}
+
+// Implements io.ReadCloser interface
+func (bf bufReader) Close() error {
```

Comment:
This is to satisfy the golang http.Request body, which is an io.ReadCloser.

See https://golang.org/pkg/net/http/#Request